tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/NUnit/NUnitWorkItemPerformWorkIntegration.cs

```diff
@@ -0,0 +1,63 @@
+// <copyright file="NUnitWorkItemPerformWorkIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.ComponentModel;
+using Datadog.Trace.Ci;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.DuckTyping;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Testing.NUnit;
+
+/// <summary>
+/// NUnit.Framework.Internal.Execution.WorkItem.PerformWork() calltarget instrumentation
+/// </summary>
+[InstrumentMethod(
+    AssemblyName = "nunit.framework",
+    TypeName = "NUnit.Framework.Internal.Execution.WorkItem",
+    MethodName = "PerformWork",
+    ReturnTypeName = ClrNames.Void,
+    MinimumVersion = "3.0.0",
+    MaximumVersion = "3.*.*",
+    IntegrationName = NUnitIntegration.IntegrationName,
+    CallTargetIntegrationType = IntegrationType.Derived)]
+[Browsable(false)]
+[EditorBrowsable(EditorBrowsableState.Never)]
+public static class NUnitWorkItemPerformWorkIntegration
+{
+    /// <summary>
+    /// OnMethodBegin callback
+    /// </summary>
+    /// <typeparam name="TTarget">Type of the target</typeparam>
+    /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+    /// <returns>Calltarget state value</returns>
+    internal static CallTargetState OnMethodBegin<TTarget>(TTarget instance)
+        where TTarget : IWorkItem
+    {
+        if (NUnitIntegration.IsEnabled)
+        {
+            var item = instance.Test;
+            switch (item.TestType)
+            {
+                case "Assembly" when item.Instance.TryDuckCast<ITestAssembly>(out var itemAssembly):
```

Comment:
If `ITestAssembly` _can_ be a `struct`, there's another API you can use here IIRC which supports `TryDuckCast` on structs

Reply:
true, done in [003ba63](https://github.com/DataDog/dd-trace-dotnet/pull/3276/commits/003ba63c2cb232e90de95c46e5f76f84ce738d3b)